scalding-core/src/main/scala/com/twitter/scalding/ExecutionOptimizationRules.scala

```diff
@@ -108,24 +109,368 @@ object ExecutionOptimizationRules {
         }
     }
 
-  object ZipWrite extends PartialRule[Execution] {
-    type InputType = (Config, Mode, Execution.Writer, ConcurrentExecutionContext)
-    type ResFun[T] =  InputType => Future[T]
+  sealed abstract class HList
```

Comment:
I didn't want to add a dependency to shapeless just to get a simple HList, so I implemented this one.